design-system/react/src/components/InputAmount/InputAmount.tsx

```diff
@@ -73,57 +82,97 @@ export const InputAmount: InputAmountComponent = ({
     }
   };
 
+  const tokenImage = (name?: string, imageUrl?: string) => {
+    return (
+      <Image
+        alt={name}
+        src={imageUrl}
+        width={14}
+        height={14}
+        css={styles.image}
+      />
+    );
+  };
+
   return (
     <Input size="lg" css={styles.input} {...props}>
-      <Input.Number
-        autoComplete="off"
-        inputMode="decimal"
-        name={name}
-        aria-label={name}
-        placeholder="0.00"
-        allowedDecimalSeparators={['.', ',']}
-        allowNegative={false}
-        thousandSeparator={false}
-        value={assetAmount}
-        onChange={(e) => {
-          handleAmountChange(e.target.value);
-        }}
-        decimalScale={units}
-        {...inputProps}
-      />
-      {initialBalance && (
-        <Input.ElementRight css={styles.elementRight}>
-          <Box css={styles.balanceActions}>
-            {!hiddenMaxButton && (
-              <Flex align="center" justify="end">
+      <Text color="intentsBase9" css={styles.heading}>
+        Amount
+      </Text>
+      <Flex align="center" css={styles.secondRow}>
+        <Input.Number
+          autoComplete="off"
+          inputMode="decimal"
+          name={name}
+          aria-label={name}
+          placeholder="0.00"
+          allowedDecimalSeparators={['.', ',']}
+          allowNegative={false}
+          thousandSeparator={false}
+          value={assetAmount}
+          onChange={(e) => {
+            handleAmountChange(e.target.value);
+          }}
+          decimalScale={units}
+          {...inputProps}
+        />
+
+        {initialBalance && (
+          <Input.ElementRight css={styles.elementRight}>
+            <Box css={styles.balanceActions}>
+              {!hiddenMaxButton && (
+                <Flex align="center">
+                  <Button
+                    aria-label="Max"
+                    variant="solid"
+                    intent="primary"
+                    onPress={handleSetBalance}
+                    css={styles.maxButton}
+                  >
+                    MAX
+                  </Button>
+                </Flex>
+              )}
+              {asset && onClickAsset && (
                 <Button
-                  aria-label="Max"
-                  size="sm"
-                  variant="ghost"
-                  onPress={handleSetBalance}
-                  css={styles.maxButton}
+                  size="xs"
+                  aria-label="Coin Selector"
+                  variant="outlined"
+                  intent="base"
+                  css={styles.assetButton}
+                  onPress={onClickAsset}
                 >
-                  Max
+                  {tokenImage(asset.name, asset.imageUrl)}
+                  <Text fontSize="xs" color="intentsBase12">
```

Comment:
Always avoid props to define css, instead use always the `style.*` to create definitions